pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -131,6 +180,45 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Is the user banned? To determine this we query the BannedUser resource for any matching entries.  The query
+	// is based on the user's emailHash value - if there is a match, and the e-mail addresses are equal, then the
+	// user is banned.
+	banned := false
+
+	// Lookup the user email label
+	if emailLbl, exists := instance.Labels[toolchainv1alpha1.UserSignupUserEmailAnnotationKey]; exists {
```

Comment:
Let's maybe move the entire banned check to a separate method?
So we can do something like:

```
banned, err := isBanned(...)
if err != nil {
    return reconcile.Result{}, err
}

Reply:
I tried this, however there are too many places within the ban check where the function returns.

Reply:
But all these returns are supposed to return an error, right? Or I'm missing something?

Reply:
https://github.com/codeready-toolchain/host-operator/pull/142#pullrequestreview-353435994

Reply:
They won't all return an error, many of them will simply update the `UserSignup` status (even if it's an invalid status) and then return.

Reply:
They should return errors (see my other comment) but even if they didn't then returning no error means returning nil :) So it still should work in both cases.